testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1314,12 +1316,18 @@ impl BitcoinRegtestController {
                 .unwrap();
             let canonical_sortition_tip =
                 SortitionDB::get_canonical_burn_chain_tip(self.sortdb_ref().conn()).unwrap();
-            if canonical_burnchain_tip.block_height == canonical_sortition_tip.block_height {
+
+            if canonical_burnchain_tip.block_height == canonical_sortition_tip.block_height
```

Comment:
What are these values on each loop pass?  Like, why does the timeout break out of this loop? Would love some more insight on why it is never the case on restart that `canonical_burnchain_tip.block_height == canonical_sortition_tip.block_height`.

Reply:
This is the scenario that I _think_ this is catching, but should perhaps be caught differently:

1. A burn block is processed by the burnchain db
2. Coordinator is notified
3. Before coordinator processes the sortition, the node exits gracefully

On restart:
1. The neon runloop syncs the burnchain, starts the coordinator (but _does not_ tell it to try to process burn blocks or stacks blocks)
2. The neon runloop invokes `wait_for_sortitions`, which will wait for the sortition db to catch up to the burnchain db. However, that won't happen until the coordinator is told to process a new burn block.

I _think_ this stall would be pushed forward by the arrival of a new burn block. An alternative solution would be for the neon runloop to just tell the chains coordinator to wake up and try to process a burn block. I think that approach would be better: the issue isn't that there's a stall, the issue is that the sortition db hasn't caught up with the burnchain db.

Reply:
@kantai yes you're right, we are looking at the use case that you've described. And when the stacks-node is being interrupted when the burnchain block pipeline is being fed, chances that you end up in this situations are high.

Reply:
Okay -- in that case, I'd suggest addressing the issue by issuing a `new_burn_block` notification to the coordinator rather than adding a timeout here.

Reply:
I can confirm that this new approach is working. Thank you for that tip @kantai!